rocAL/rocAL/include/meta_data/meta_data.h

```diff
@@ -153,78 +235,129 @@ struct MetaDataBatch
     virtual void clear() = 0;
     virtual void resize(int batch_size) = 0;
     virtual int size() = 0;
+    virtual void copy_data(std::vector<void*> buffer) = 0;
+    virtual std::vector<size_t>& get_buffer_size() = 0;
     virtual MetaDataBatch&  operator += (MetaDataBatch& other) = 0;
     MetaDataBatch* concatenate(MetaDataBatch* other)
     {
         *this += *other;
         return this;
     }
-    virtual std::shared_ptr<MetaDataBatch> clone()  = 0;
-    std::vector<int>& get_label_batch() { return _label_id; }
-    std::vector<BoundingBoxCords>& get_bb_cords_batch() { return _bb_cords; }
-    std::vector<BoundingBoxCords_xcycwh>& get_bb_cords_batch_xcycxwh() { return _bb_cords_xcycwh; }
-    std::vector<BoundingBoxLabels>& get_bb_labels_batch() { return _bb_label_ids; }
-    ImgSizes & get_img_sizes_batch() { return _img_sizes; }
-    JointsDataBatch & get_joints_data_batch() { return _joints_data; }
+    virtual std::shared_ptr<MetaDataBatch> clone() = 0;
+    virtual void get_labels_batch(std::vector<Labels>** label_ids) {}
+    virtual void get_bb_cords_batch(std::vector<BoundingBoxCords>** bb_cords) {}
+    virtual void get_bb_cords_batch_xcycxwh(std::vector<BoundingBoxCords_xcycwh>** bb_cords_xcycwh) {}
```

Comment:
remove bb_cords_batch_xywh and use same function for both. use flag to differentiate between ltrb or xywh

Reply:
Done